x-pack/plugins/code/server/lsp/java_launcher.ts

```diff
@@ -155,4 +167,70 @@ export class JavaLauncher implements ILanguageServerLauncher {
     );
     return p;
   }
+
+  private checkJavaRuntime() {
```

Comment:
change name to getSystemJavaHome(): String

in mac, you should also run java_home to get list of java homes

but I don't think it worth doing that, but you could add a todo